iceoryx_hoofs/include/iceoryx_hoofs/log/platform_building_blocks/console_logger.hpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 - 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_PLATFORM_BUILDING_BLOCKS_CONSOLE_LOGGER_HPP
+#define IOX_HOOFS_PLATFORM_BUILDING_BLOCKS_CONSOLE_LOGGER_HPP
+
+#include "iceoryx_hoofs/log/platform_building_blocks/logcommon.hpp"
+
+#include <atomic>
+#include <cstdint>
+#include <mutex>
+
+namespace iox
+{
+namespace pbb
+{
+class ConsoleLogger
```

Comment:
@elfenpiff I have to think about this when I'm rested :)

Reply:
@elfenpiff I think I messed a little bit up when I tried the different approaches.

1. Your might be right that `m_activeLogLevel` does not belong into `ConsoleLogger`. I'm thinking whether it makes sense to move it into `Logger` but since `Logger` cannot be changed by the user and I wanted to give them some freedom on how to implement the active log level I decided to move it into the logger implementation which by default is the `ConsoleLogger`. This means also that I just cannot rid of `setLogLevel` and `getLogLevel` since this will be the user API. Following your suggestion would mean to create an additional struct with only one static member. What do you think of renaming `m_activeLogLevel` to `s_activeLogLevel` to make it more clear that it is a static variable?

2. Done in #1621 

3. Almost. There would be the `s_activeLogLevel`

Reply:
Okay lets rename it into `s_activeLogLevel`.